components/org.wso2.carbon.identity.oauth.endpoint/src/test/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpointTest.java

```diff
@@ -221,36 +328,238 @@ public Object answer(InvocationOnMock invocation) throws Throwable {
+                                                       String responseMode, String redirectUri, int expected)
+            throws Exception {
+        mockStatic(SessionDataCache.class);
+        when(SessionDataCache.getInstance()).thenReturn(sessionDataCache);
+        SessionDataCacheKey loginDataCacheKey = new SessionDataCacheKey(this.sessionDataKey);
+        when(sessionDataCache.getValueFromCache(loginDataCacheKey)).thenReturn(loginCacheEntry);
+
+        AuthenticationResult result =
+                setAuthenticationResult(isAuthenticated, attributes, errorCode, errorMsg, errorUri);
+
+        AuthenticationResult resultInRequest = null;
+        AuthenticationResultCacheEntry authResultCacheEntry = null;
+        if (isResultInRequest) {
+            resultInRequest = result;
+        } else {
+            authResultCacheEntry = new AuthenticationResultCacheEntry();
+            authResultCacheEntry.setResult(result);
         }
-        requestParams.put("sessionDataKeyConsent", new String[]{sessionDataKayConsent});
-        requestParams.put("tocommonauth", new String[]{toCommonAuth});
-        requestParams.put("scope", new String[]{scope});
 
-        final Map<String, Object> requestAttributes = new HashedMap();
-        requestAttributes.put("authenticatorFlowStatus", flowStatus);
-        requestAttributes.put("sessionDataKey", sessionDataKey);
+        Map<String, String[]> requestParams = new HashedMap();
+        Map<String, Object> requestAttributes = new HashedMap();
+
+        requestParams.put(CLIENT_ID, new String[]{clientId});
+        requestParams.put(FrameworkConstants.RequestParams.TO_COMMONAUTH, new String[]{"false"});
+        requestParams.put(OAuthConstants.OAuth20Params.SCOPE, new String[]{"openid"});
+
+        requestAttributes.put(FrameworkConstants.RequestParams.FLOW_STATUS, AuthenticatorFlowStatus.INCOMPLETE);
+        requestAttributes.put(FrameworkConstants.SESSION_DATA_KEY, sessionDataKey);
+        requestAttributes.put(FrameworkConstants.RequestAttribute.AUTH_RESULT, resultInRequest);
+
+        mockHttpRequest(requestParams, requestAttributes);
+
+        mockStatic(FrameworkUtils.class);
+        when(FrameworkUtils.getAuthenticationResultFromCache(anyString())).thenReturn(authResultCacheEntry);
+
+        OAuth2Parameters oAuth2Params = setOAuth2Parameters(scopes, appName, responseMode, redirectUri);
+        when(loginCacheEntry.getoAuth2Parameters()).thenReturn(oAuth2Params);
+        when(loginCacheEntry.getLoggedInUser()).thenReturn(result.getSubject());
+
+        mockStatic(OAuthServerConfiguration.class);
+        when(OAuthServerConfiguration.getInstance()).thenReturn(oAuthServerConfiguration);
+        when(oAuthServerConfiguration.getPersistenceProcessor()).thenReturn(tokenPersistenceProcessor);
+        when(tokenPersistenceProcessor.getProcessedClientId(anyString())).thenAnswer(new Answer<Object>(){
+            @Override
```

Comment:
Shall we use do call real ?

Reply:
Need one more level of mocking to call the real method since tokenPersistenceProcessor is mocking an interface.